File: CPP/FilmAnimasi.cpp
using namespace std;    // pakai namespace std (tanpa prefiks std::)

class FilmAnimasi : public FilmBioskop {  // deklarasi class (level 3), turunan dari FilmBioskop
private:
    string studio;                        // atribut 1: studio animasi pembuat film
    string teknikAnimasi;                 // atribut 2: teknik animasi (2D / 3D / stop motion)
    string ratingUsia;                    // atribut 3: rating usia penonton (SU / 13+ / 17+)

public:
    // FilmAnimasi: Konstruktor untuk menyimpan data lengkap film animasi.
    FilmAnimasi(int id, string judul, int harga, int durasiMenit,vector<string> genre, string sutradara,
                string studio, string teknikAnimasi, string ratingUsia)
                : FilmBioskop(id, judul, harga, durasiMenit, genre, sutradara) {
        this->studio        = studio;         // simpan nama studio pembuat
        this->teknikAnimasi = teknikAnimasi;  // simpan teknik animasi yang dipakai
        this->ratingUsia    = ratingUsia;     // simpan rating usia penonton
    }

    // ===== getter =====
    // getStudio: Getter untuk mengambil atribut studio.
    string getStudio()        { return studio; }         // getter studio animasi
    // getTeknikAnimasi: Getter untuk mengambil atribut teknikAnimasi.
    string getTeknikAnimasi() { return teknikAnimasi; }  // getter teknik animasi
    // getRatingUsia: Getter untuk mengambil atribut ratingUsia.
    string getRatingUsia()    { return ratingUsia; }     // getter rating usia

    // ===== setter =====
    // setStudio: Setter untuk mengubah atribut studio.
    void setStudio(string studio)              { this->studio = studio; }                 // setter studio animasi
    // setTeknikAnimasi: Setter untuk mengubah atribut teknikAnimasi.
    void setTeknikAnimasi(string teknikAnimasi) { this->teknikAnimasi = teknikAnimasi; }  // setter teknik animasi
    // setRatingUsia: Setter untuk mengubah atribut ratingUsia.
    void setRatingUsia(string ratingUsia)      { this->ratingUsia = ratingUsia; }         // setter rating usia

    // getData: Mengambil seluruh data Film, bioskop, dan animasi (9 kolom).
    vector<string> getData() {                         // method getData milik FilmAnimasi (menutup milik FilmBioskop)
        vector<string> data = FilmBioskop::getData();  // ambil 6 kolom dari class induk
        data.push_back(studio);                        // tambah kolom studio animasi
        data.push_back(teknikAnimasi);                 // tambah kolom teknik animasi
        data.push_back(ratingUsia);                    // tambah kolom rating usia
        return data;                                   // kembalikan 9 kolom gabungan
    }
};

File: Java/FilmAnimasi.java
// =====================================================================
// Class level 3 (turunan dari FilmBioskop). 
// =====================================================================
public class FilmAnimasi extends FilmBioskop {  // deklarasi class level 3: turunan dari FilmBioskop
    private String studio;                      // atribut nama studio pembuat animasi
    private String teknikAnimasi;               // atribut teknik animasi yang dipakai
    private String ratingUsia;                  // atribut rating usia penonton

    // FilmAnimasi: Konstruktor untuk menginisialisasi seluruh atribut berantai tiga level class.
    public FilmAnimasi(int id, String judul, int harga, int durasiMenit,     // konstruktor: terima seluruh atribut 
                       java.util.ArrayList<String> genre, String sutradara,  // lanjutan parameter: genre, sutradara, studio
                       // parameter lanjutan: teknik animasi dan rating usia
                       String studio, String teknikAnimasi, String ratingUsia) {
        super(id, judul, harga, durasiMenit, genre, sutradara);           // panggil konstruktor class induk (FilmBioskop)
        this.studio = studio;                                             // simpan studio animasi ke atribut
        this.teknikAnimasi = teknikAnimasi;                               // simpan teknik animasi ke atribut
        this.ratingUsia = ratingUsia;                                     // simpan rating usia ke atribut
    }

    // ===== getter =====
    // getStudio: Mengembalikan nama studio animasi.
    public String getStudio() {  // getter: mengembalikan nama studio
        return studio;           // kembalikan nama studio animasi
    }

    // getTeknikAnimasi: Mengembalikan teknik animasi yang dipakai.
    public String getTeknikAnimasi() {  // getter: mengembalikan teknik animasi
        return teknikAnimasi;           // kembalikan teknik animasi yang dipakai
    }

    // getRatingUsia: Mengembalikan rating usia penonton.
    public String getRatingUsia() {  // getter: mengembalikan rating usia
        return ratingUsia;           // kembalikan rating usia penonton
    }

    // ===== setter =====
    // setStudio: Mengubah nama studio animasi.
    public void setStudio(String studio) {  // setter: mengubah nama studio
        this.studio = studio;               // simpan nama studio baru ke atribut
    }

    // setTeknikAnimasi: Mengubah teknik animasi.
    public void setTeknikAnimasi(String teknikAnimasi) {  // setter: mengubah teknik animasi
        this.teknikAnimasi = teknikAnimasi;               // simpan teknik animasi baru ke atribut
    }

    // setRatingUsia: Mengubah rating usia penonton.
    public void setRatingUsia(String ratingUsia) {  // setter: mengubah rating usia
        this.ratingUsia = ratingUsia;               // simpan rating usia baru ke atribut
    }

    // getData: Mengembalikan seluruh data gabungan tiga level class (sembilan kolom).
    public java.util.ArrayList<String> getData() {           // method milik FilmAnimasi (menutup milik FilmBioskop)
        java.util.ArrayList<String> data = super.getData();  // ambil dulu data milik class induk (6 kolom)
        data.add(studio);                                    // kolom studio animasi
        data.add(teknikAnimasi);                             // kolom teknik animasi
        data.add(ratingUsia);                                // kolom rating usia
        return data;                                         // kembalikan sembilan kolom data lengkap
    }
}